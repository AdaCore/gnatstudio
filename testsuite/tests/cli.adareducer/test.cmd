echo "## TEST HELP"
gnatstudio_cli adareducer --help | sed -e "s/.exe//"
echo ""

echo "## TEST MISSING PROJECT"
gnatstudio_cli adareducer
echo ""

echo "## TEST MISSING ORACLE"
gnatstudio_cli adareducer -P p
echo ""

echo "## TEST FUNCTIONALITY"
gnatstudio_cli adareducer -P p -s oracle.sh > /dev/null
cat hello.adb
