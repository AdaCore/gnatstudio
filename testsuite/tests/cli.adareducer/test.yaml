title: 'cli.adareducer'
#  the test fails nightly on Windows while working locally
skip:
    - ['SKIP', 'env.build.os.name == "windows"']