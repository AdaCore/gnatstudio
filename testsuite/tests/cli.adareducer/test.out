## TEST HELP
GNAT Studio command line interface (adareducer mode)
Usage: gnatstudio_cli [switches] [arguments]

 -P, --project=ARG       Load the given project (mandatory)
 --single_file=ARG       Specify the base name of a file where processing should start.
 --follow-closure        Follow the closure of the file
 -s, --oracle_script=ARG The reduction oracle script.

## TEST MISSING PROJECT
Specify a project with the -P switch

## TEST MISSING ORACLE
Specify an oracle with the -s/--oracle_script switch

## TEST FUNCTIONALITY
procedure Hello is
begin
   null;
end Hello;